exonum/src/blockchain/service.rs

```diff
@@ -395,19 +398,42 @@ impl SharedNodeState {
     }
     /// Updates internal state, from `State` of a blockchain node.
     pub fn update_node_state(&self, state: &State) {
-        self.state
-            .write()
-            .expect("Expected write lock.")
-            .peers_info
-            .clear();
+        let mut lock = self.state.write().expect("Expected write lock.");
+        lock.peers_info.clear();
+        lock.majority_count = state.majority_count();
+        lock.is_validator = state.is_validator();
+        lock.validators.clear();
 
         for (p, c) in state.peers().iter() {
-            self.state
-                .write()
-                .expect("Expected write lock.")
-                .peers_info
-                .insert(c.addr(), *p);
+            lock.peers_info.insert(c.addr(), *p);
         }
+        for v in state.validators().iter() {
+            lock.validators.push(*v);
+        }
+    }
+
+    /// Returns a boolean value which indicates whether the consensus is
+    /// Returns the majority count from the current "State"
+    /// of a blockchain node.
+    pub fn is_consensus(&self) -> bool {
+        let lock = self.state.read()
+            .expect("Expected read lock.");
+
+        let mut active_validators :usize = 0;
+        for (_s, p) in lock.peers_info.iter() {
+            if lock.validators.iter()
+                .any(|x| x.consensus_key == *p) {
+                active_validators = active_validators.saturating_add(1);
```

Comment:
I like safety, but `active_validators` is `usize` and I don't think it's actually realistic to use `saturating_add`.